File: src/0001_two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {num: idx for idx, num in enumerate(nums)}

        for idx, num in enumerate(nums):
            if (lookup_num := target - num) in lookup:
                if (lookup_idx := lookup[lookup_num]) != idx:
                    return [idx, lookup_idx]

        return None

File: Elixir/0001_two-sum.ex
defmodule Solution do
  @spec two_sum(nums :: [integer], target :: integer) :: [integer]
  def two_sum(nums, target) do
    Enum.with_index(nums)
    |> Enum.reduce_while(%{}, fn {num, idx}, acc ->
      to_find = target - num

      case Map.has_key?(acc, to_find) do
        true -> {:halt, [acc[to_find], idx]}
        false -> {:cont, Map.put(acc, num, idx)}
      end
    end)
  end
end
